File: app/src/main/java/dk/itu/shoppingv1/ShoppingActivity.java
package dk.itu.shoppingv1;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class ShoppingActivity extends AppCompatActivity {

    //Shopping V1

    // Model: Database of items
    private ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.shopping);


        itemsDB= new ItemsDB();
        TextView responseText = findViewById(R.id.textView1);
        EditText edit = findViewById(R.id.edit_text);


        Button checkItem= findViewById(R.id.list_button);
        checkItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String response = itemsDB.whereDoesThisGo(edit.getText().toString());
                responseText.setText("Those go in "+response);
            }
        });
    }
}

File: GarbageSortingW4/app/src/main/java/dk/itu/myshoppingv3kotlin/ShoppingActivity.kt
package dk.itu.myshoppingv3kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import android.widget.Toast


class ShoppingActivity : AppCompatActivity() {
    // Model: Database of items
    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.shopping)

        //getting the items
        itemsDB = ItemsDB.get() //this works as a constructor it seems

        val intent = intent

        // Get the data from the Intent extras using the same key you used in the sending activity
        val newdata:String? = intent.getStringExtra("newItem")
        val newplace:String? = intent.getStringExtra("newPlace")

        if(newdata!=null && newplace!=null){
            itemsDB.addItem(newdata, newplace)
            Log.d("newData", newdata)
        }

        //Input text field
        val inputTextField: TextView = findViewById(R.id.what_text)

        //Response from the system


        val listItems: Button = findViewById(R.id.list) //
        listItems.setOnClickListener {
            //what was written
            val inputFromUser :String = inputTextField.text.toString()

            //here get the response from the database
            val responseFromDB:String? = itemsDB.whereDoesItGo(inputFromUser)

            //fill in the response box
            //if(response)
            val responseBox :TextView = findViewById(R.id.textResponse)
            responseBox.text = "You should throw the $inputFromUser in the $responseFromDB"
        }

        val addItem: Button = findViewById(R.id.add_button)
        addItem.setOnClickListener {
            Log.d("ButtonClickListener", "Button clicked!")
            //val myIntent = intent(this, AddingActivity::class.java)
            val intent = Intent(this, AddingActivity::class.java)
            startActivity(intent)
        }
    }
}